monitoring/serf.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2016 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"sync"
+
+	"github.com/gravitational/trace"
+	serf "github.com/hashicorp/serf/client"
+)
+
+// serfClient is the minimal interface to the serf cluster.
+// It enables mocking access to serf network in tests.
+type serfClient interface {
```

Comment:
Why do you need this? I would export the bits from the agent package so these could be used if necessary.

Reply:
I was thinking about that but in order to do that, shouldn't I change `serfClient` to be `SerfClient` thus needing a refactoring in all the `agent` package where that's used?

Is that okay?

Reply:
done.